src/osd/osd_types.cc

```diff
@@ -694,6 +694,20 @@ void coll_t::generate_test_instances(list<coll_t*>& o)
 
 // ---
 
+std::string pg_vector_string(const vector<int32_t> &a)
+{
+  ostringstream oss;
+  oss << "[";
+  for (vector<int32_t>::const_iterator i=a.begin(); i!=a.end(); i++) {
+    if (i!=a.begin()) oss << ",";
+    if (*i!=CRUSH_ITEM_NONE) oss << *i;
+    else oss << "CRUSH_ITEM_NONE";
```

Comment:
I think just "NONE" would be sufficient?

Also, can you please add spaces around the = != and put the non-conditional part of the if's on new lines?

<code>
 if (foo)
    bar;
 else
    baz;
</pre>
(Gory details in CodingStyle.)  Otherwise, nice cleanup, thanks!  Thanks!